src/dsp_tools/commands/xml_validate/reformat_input.py

```diff
@@ -0,0 +1,194 @@
+from typing import Callable
+from typing import Sequence
+
+from lxml import etree
+from rdflib import XSD
+from rdflib import Literal
+from rdflib import URIRef
+
+from dsp_tools.commands.xml_validate.models.data_rdf import BooleanValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import ColorValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import DataRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import DateValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import DecimalValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import GeonameValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import IntValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import LinkValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import ListValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import ResourceRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import RichtextRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import SimpleTextRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import TimeValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import UriValueRDF
+from dsp_tools.commands.xml_validate.models.data_rdf import ValueRDF
+
+
+def transform_into_data_deserialised(root: etree._Element) -> DataRDF:
+    """
+    Takes the root of an XML
+    Extracts the metadata of the project and transforms all its resources.
+
+    Args:
+        root: root of an xml with qnames and comments removed
+
+    Returns:
+        Class instance with the information reformatted
+    """
+    shortcode = root.attrib["shortcode"]
+    default_ontology = root.attrib["default-ontology"]
+    resources = _deserialise_all_resources(root)
+    return DataRDF(shortcode=shortcode, default_onto=default_ontology, resources=resources)
+
+
+def _deserialise_all_resources(root: etree._Element) -> list[ResourceRDF]:
+    all_res = []
+    for res in root.iterchildren():
+        match res.tag:
+            case "resource":
+                all_res.append(_deserialise_one_resource(res))
+            case "annotation" | "region" | "link" | "video-segment" | "audio-segment":
+                all_res.append(_deserialise_dsp_base_resource(res))
+            case _:
+                pass
```

Comment:
I would rather raise an error here. This is a critical error, which we shouldn't miss.
The permissions-elements can be passed, of course:

```suggestion
            case "permissions":
                pass
            case _:
                raise BaseError()
```

Reply:
No I don't want to do that, because there are permissions we want to ignore. Since it passed our xml validation, we can trust the file